File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/MarketDataRequest/ElementMarketSubscriptionCanceled.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.MarketDataRequest.ElementMarketSubscriptionCanceled.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.MarketDataRequest;

import com.bemu.BEmu.Name;

public class ElementMarketSubscriptionCanceled extends ElementParent
{
    private final ElementMarketSubscriptionCancelReason _reason;

    ElementMarketSubscriptionCanceled(MessageMarketSubscriptionCanceled arg) throws Exception
    {
        this._reason = (ElementMarketSubscriptionCancelReason)arg.getElement("reason");
    }
	
    public Name name()
    {
    	return new Name("SubscriptionTerminated");
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 1;
    }
    
    public boolean isComplexType()
    {
    	return true;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
        String tabs = com.bemu.BEmu.types.IndentType.Indent(tabIndent);
        StringBuilder result = new StringBuilder();

        result.append(String.format("%s%s = {%s", tabs, this.name().toString(), System.getProperty("line.separator")));
        result.append(this._reason.prettyPrint(tabIndent + 1));
        result.append(String.format("%s}%s", tabs, System.getProperty("line.separator")));

        return result;
    }
    
    
    
    
}

File: BEmu_cpp/src/MarketDataRequest/ElementMarketSubscriptionCanceled.cpp
//------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/MarketDataRequest/ElementMarketSubscriptionCanceled.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

#include "MarketDataRequest/ElementMarketSubscriptionCanceled.h"
#include "MarketDataRequest/MessageMarketSubscriptionCanceled.h"
#include "MarketDataRequest/ElementMarketSubscriptionCancelReason.h"

#include "Types/IndentType.h"

namespace BEmu
{
	namespace MarketDataRequest
	{
		ElementMarketSubscriptionCanceled::ElementMarketSubscriptionCanceled(MessageMarketSubscriptionCanceled * arg)
		{
			this->_reason = (ElementMarketSubscriptionCancelReason*)arg->getElement("reason");
		}

		Name ElementMarketSubscriptionCanceled::name() const
		{
			return Name("SubscriptionTerminated");
		}

		size_t ElementMarketSubscriptionCanceled::numValues() const
		{
			return 1;
		}

		size_t ElementMarketSubscriptionCanceled::numElements() const
		{
			return 1;
		}

		bool ElementMarketSubscriptionCanceled::isNull() const
		{
			return false;
		}

		bool ElementMarketSubscriptionCanceled::isArray() const
		{
			return false;
		}

		bool ElementMarketSubscriptionCanceled::isComplexType() const
		{
			return true;
		}

		std::ostream& ElementMarketSubscriptionCanceled::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			std::string tabs(IndentType::Indent(level, spacesPerLevel));

			stream << tabs << "SubscriptionTerminated = {" << std::endl;
			this->_reason->print(stream, level + 1, spacesPerLevel);
			stream << tabs << '}' << std::endl;

			return stream;
		}
	}
}